tracee-ebpf/tracee/tracee.go

```diff
@@ -1108,6 +1152,29 @@ func (t *Tracee) processEvent(ctx *context, args map[argTag]interface{}) error {
 	return nil
 }
 
+func getFileHash(fileName string) string {
```

Comment:
UpdateFileSHA, and thus this function as well, is called on terminating tracee.
At this point in time, sourceFilePath, which is given as an argument to updateProfile and is composed from a path in "/proc/pid/root/..." might not be available.
Instead, as we already capture the executed files to destinationFilePath), we can perform the hash calculations on these captured files

Reply:
yes this was the intention with https://github.com/aquasecurity/tracee/pull/725#discussion_r637391823

Reply:
Are you sure?
`updateProfile()` is being called as follows:
`t.updateProfile(fmt.Sprintf("%s:%d", capturedFileID, sourceFileCtime))`
were capturedFileID equals:
`fmt.Sprintf("%d:%s", ctx.MntID, sourceFilePath)`
so in total we get:
`t.updateProfile(fmt.Sprintf(""%d:%s:%d", ctx.MntID, sourceFilePath, sourceFileCtime))`
Which means that sourceFilePath is saved with a value of "/proc/pid/root/...", isn't it?

Reply:
yes, I wrote this was the _intention_ of the comment I made. I didn't see it implemented yet.

Reply:
I added some logic that uses captured file info instead of using the "/proc/pid/root" files. https://github.com/aquasecurity/tracee/pull/725/commits/25c83dfb295b1346c7cb2c040196fe4dcd7a18b0